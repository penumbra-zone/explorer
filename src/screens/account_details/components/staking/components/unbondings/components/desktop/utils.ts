export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'commission',
    align: 'right',
    width: 25,
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'linkedUntil',
    width: 25,
    align: 'right',
  },
];
