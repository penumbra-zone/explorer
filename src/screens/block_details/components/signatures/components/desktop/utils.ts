export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 34,
  },
  {
    key: 'votingPower',
    align: 'right',
    width: 33,
  },
  {
    key: 'signed',
    width: 33,
  },
];
