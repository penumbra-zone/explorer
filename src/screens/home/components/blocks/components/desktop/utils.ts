export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
}[] = [
  {
    key: 'height',
  },
  {
    key: 'validator',
  },
  {
    key: 'hash',
  },
  {
    key: 'txs',
  },
  {
    key: 'time',
    align: 'right',
  },
];
