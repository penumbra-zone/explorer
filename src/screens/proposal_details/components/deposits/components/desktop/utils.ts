export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 34,
  },
  {
    key: 'amount',
    width: 33,
    align: 'right',
  },
  {
    key: 'time',
    width: 33,
  },
];
