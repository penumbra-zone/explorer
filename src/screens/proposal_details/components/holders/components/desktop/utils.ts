export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 10,
  },
  {
    key: 'address',
    width: 30,
  },
  {
    key: 'quantity',
    width: 20,
    align: 'right',
  },
  {
    key: 'percentage',
    width: 20,
  },
  {
    key: 'value',
    align: 'right',
    width: 20,
  },
];
