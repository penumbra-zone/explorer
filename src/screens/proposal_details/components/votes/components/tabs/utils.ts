export const tabLabels = [
  {
    key: 'all',
    num: 30,
  },
  {
    key: 'yes',
    num: 3,
  },
  {
    key: 'no',
    num: 3,
  },
  {
    key: 'noWithVeto',
    num: 3,
  },
];
