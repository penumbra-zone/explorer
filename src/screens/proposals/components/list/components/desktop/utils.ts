export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'idx',
    width: 15,
  },
  {
    key: 'token',
    width: 10,
  },
  {
    key: 'price',
    align: 'right',
    width: 15,
  },
  {
    key: 'change',
    align: 'right',
    width: 15,
  },
  {
    key: 'volume',
    align: 'right',
    width: 15,
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 15,
  },
  {
    key: 'holders',
    align: 'right',
    width: 15,
  },
];
