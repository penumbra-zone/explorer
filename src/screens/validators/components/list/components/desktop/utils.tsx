import React from 'react';
import {
  Typography,
} from '@material-ui/core';
import { InfoPopover } from '@components';

export const fetchColumns = (t): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sort?: boolean;
}[] => {
  return ([
    {
      key: 'idx',
      width: 10,
    },
    {
      key: 'validator',
      width: 20,
      sort: true,
    },
    {
      key: 'stake',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'fee',
      align: 'right',
      width: 10,
      sort: true,
    },
    {
      key: 'lastVote',
      align: 'right',
      width: 15,
    },
    {
      key: 'skipRate',
      width: 15,
      sort: true,
    },
    {
      key: 'condition',
      align: 'center',
      width: 15,
      component: (
        <Typography variant="h4" className="label popover">
          {t('condition')}
          <InfoPopover
            content={t('conditionExplanation')}
          />
        </Typography>
      ),
    },
  ]);
};
