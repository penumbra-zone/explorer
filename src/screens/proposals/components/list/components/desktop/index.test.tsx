import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Desktop from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Loading: (props) => <div id="Loading" {...props} />,
}));

jest.mock('react-virtualized-auto-sizer', () => ({ children }: any) => children({
  height: 600, width: 600,
}));

jest.mock('@src/screens/blocks/components/list/contexts/blocks', () => ({
  useBlocksContext: () => ({
    items: [{
      token: 'BTC',
      price: '$1,802',
      change: 10,
      volume: '$15,902,496,558',
      marketCap: '$515,912,496',
      holders: '2,178,994',
      address: '9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E',
    }],
    itemsCount: 2,
    loadMoreItems: jest.fn(),
    isItemLoaded: jest.fn(),
  }),
}));

// ==================================
// unit tests
// ==================================
describe('screen: Tokens/Desktop', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Desktop />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
