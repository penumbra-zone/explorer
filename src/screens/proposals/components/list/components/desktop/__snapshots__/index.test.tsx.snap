// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Tokens/Desktop matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div
    onScroll={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 50,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        Object {
          "height": 50,
          "pointerEvents": undefined,
          "width": 600,
        }
      }
    >
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          idx
        </h4>
      </div>
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 90,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 60,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          token
        </h4>
      </div>
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 150,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          price
        </h4>
      </div>
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 240,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          change
        </h4>
      </div>
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 330,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          volume
        </h4>
      </div>
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 420,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          marketCap
        </h4>
      </div>
      <div
        className="makeStyles-cell"
        style={
          Object {
            "height": 50,
            "left": 510,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          holders
        </h4>
      </div>
    </div>
  </div>
  <div
    onScroll={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 550,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        Object {
          "height": NaN,
          "pointerEvents": undefined,
          "width": 600,
        }
      }
    />
  </div>
</div>
`;
